docs/api.md

```diff
@@ -1,8 +1,88 @@
+
+* name -- name of profile
+* namespace -- namespace from which BareMetalHosts to be fetched
+* labels -- label to be set on matching BareMetalHosts
+
+### HardwareClassificationController spec
+
+The *HardwareClassificationController's* *spec* contains mainly expected hardware configuration details.
+
+#### Spec fields
+
+* *hardwareCharacteristics* -- HardwareCharacteristics defines expected hardware configurations for CPU, DISK, NIC and RAM.
+  * *cpu* -- Expected CPU configurations:
+    * minimumCount -- minimum cpu count
+    * maximumCount -- maximum cpu count
+    * minimumSpeedMHz -- minimum speed in MHz
+    * maximumSpeedMHz -- maximum speed in MHz
+  * *disk* -- Expected DISK configurations:
+    * minimumCount -- minimum disk count
+    * maximumCount -- maximum disk count
+    * minimumIndividualSizeGB -- minimum individual disk size in GB
+    * maximumIndividualSizeGB -- maximum individual disk size in GB
+  * *ram* -- Expected RAM configurations:
+    * minimumSizeGB -- minimum ram size in GB
+    * maximumSizeGB -- maximum ram size in GB
+  * *nic* -- Expected NIC configurations:
+    * minimumCount -- minimum nic count
+    * maximumCount -- maximum nic count
+
+
+### HardwareClassificationController status
+
+The *HardwareClassificationController's* *status* which represents the observed state of HardwareClassification. 
+
+#### Status fields
+
+* *errorType* -- errorType indicates the type of failure encountered
+  * LabelUpdateFailure -- LabelUpdateFailure is an error condition occurring when the controller is unable to update label of BareMetalHost.
+  * LabelDeleteFailure -- LabelDeleteFailure is an error condition occurring when the controller is unable to delete label of BareMetalHost.
+  * FetchBMHListFailure -- FetchBMHListFailure is an error condition occurring when the controller is unable to fetch BareMetalHost from BMO.
+  * ProfileMisConfigured -- ProfileMisConfigured is an error condition occurring when the extracted profile is misconfigured.
+
+* *profileMatchStatus* -- profileMatchStatus indicates whether expected hardwareCharacteristics matches to any of BareMetalHost or not.
+  * ProfileMatchStatusEmpty -- default is empty
+  * ProfileMatchStatusMatched -- profileMatchStatusMatched is the status value when the profile matches to one of the BareMetalHost.
+  * ProfileMatchStatusUnMatched -- profileMatchStatusUnMatched is the status value when the profile does not matches to any of the BareMetalHost.
+
+* *errorMessage* -- Details of the last error reported by the hardwareclassification system.
+
+### HardwareClassificationController Example
+
+The following is a sample CRD of a HardwareClassificationController resource (in YAML), it includes its metadata and specification section.
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: HardwareClassification
+metadata:
+  name: hardwareclassification-sample
+  namespace: metal3
+  labels:
+    hardwareclassification-sample: matches
```

Comment:
Is there any place where this, and the label set on the BMH, are explained ?

Reply:
Good catch.  @nehaalhat we should add in above above explaination. How these labels added in BMH hosts and how we can see it.

Reply:
@maelk  We can mention key and value to be added on BMH hosts on one note. That's ok I guess.

Reply:
Added explanation for label parameter.
Commit: 12f5abaf4a802305ae4c068b6e2a0e871fd0b664
@digambar15 @maelk kindly review.

Reply:
Thanks!